metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -953,6 +967,54 @@ def _get_foreign_keys(
             foreign_keys = []
         return foreign_keys
 
+    def get_fine_grained_lineages(
+        self,
+        dataset_urn: str,
+        upstream_dataset_urn: str,
+        schema_fields: List[SchemaField],
+    ) -> Optional[List[FineGrainedLineage]]:
+        def simplify_field_path(field_path):
+            return Dataset._simplify_field_path(field_path)
+
+        if self.ctx.graph:
+            upstream_schema_metadata: Optional[
+                SchemaMetadata
+            ] = self.ctx.graph.get_schema_metadata(upstream_dataset_urn)
```

Comment:
Not sure if  fetching the upstream schema from the DataHub graph is an existing pattern in the connectors, so I'm asking:

* what if the upstream schema isn't published yet? should we report a warning?
* Isn't the connector itself producing the upstream schema? if so, may we skip fetching it from the DataHub graph?
* may we overloading the DataHub graph with too many requests, if we call it hundreds/thousands of times?

 Let me know your thoughts when you get a chance!

Reply:
Agree that we should avoid making graph calls like this if we can avoid it.

@acrylJonny what was the reason we need this lookup?

Reply:
Essentially it was to ensure that the upstream dataset has the same schema as the downstream. We could choose to skip this validation though.

Reply:
Yep let's just do the blind linking for now - it's what we do for most other sources that generate external lineage

The main risk there is actually column casing mismatches, but for those we'd need to use a fuzzy matcher - we recently added this to looker, but imo it's probably not necessary here